src/test/java/StringManipulationTest.java

```diff
@@ -19,63 +18,113 @@ public void tearDown() {
         manipulatedstring = null;
     }
 
+    @Test
+    public void testgetString(){
+        manipulatedstring.setString("Hello!");
+        String expected = "Hello!";
+        String actual = manipulatedstring.getString();
+        assertEquals(expected, actual);
+    }
     @Test
     public void testCount1() {
         manipulatedstring.setString("This is my string");
         int length = manipulatedstring.count();
         assertEquals(4, length);
     }
 
+    // Test when string has multiple punctuations.
```

Comment:
Testing against the punctuation is a good idea. I can see how this could easily break some implementations of the method

Reply:
You are right. If the test case was "This isn't my string !" and they put a space before the exclamation point it will break the test case because my method counts punctuations as a word too. My method implementation is way too simple to catch that. I think it would be better if I refactor that method and make it ignore punctuation.